envisage/extension_point_binding.py

```diff
@@ -23,15 +23,15 @@ class ExtensionPointBinding(HasTraits):
     #### 'ExtensionPointBinding' interface ####################################
 
     # The object that we are binding the extension point to.
-    obj = Any
+    obj = WeakRef
 
     # The Id of the extension point.
     extension_point_id = Str
 
     # The extension registry used by the binding. If this trait is not set then
     # the class-scope extension registry set on the 'ExtensionPoint' class is
     # used (and if that is not set then the binding won't work ;^)
-    extension_registry = Instance(IExtensionRegistry)
+    extension_registry = WeakRef(IExtensionRegistry)
```

Comment:
I'm also worried about unintended consequences of this change. For example, if someone creates an extension registry in an existing application, creates some bindings using that registry, and then neglects to store it anywhere, right now there will be no ill effects, since the extension registry is kept alive by the bindings. But with this change, we'd be getting a lot of TraitErrors in that situation. We may see similar effects at application tearDown, when an extension registry has been cleaned up but the bindings trying to use it are still active.